External package STI classes don't load manifests correctly (Person from smrt-profiles)

## Problem

When using STI classes from external packages (e.g., `Person`, `Organization`, `Profile` from `@happyvertical/smrt-profiles`) in a consuming package (e.g., `praeco`), the manifest loader fails to load field definitions, resulting in:

1. **`create()` returns null IDs**: `person.id` is `null` after creation
2. **Empty manifest**: Logs show "📦 Loaded manifest for external package class: Person (0 fields, 0 methods)"
3. **Records don't persist correctly**: Can't retrieve created records via `findById()` or `list()`

## Reproduction

### Test File: Person.spec.ts

```typescript
import { describe, it, expect, beforeAll } from 'vitest';
import { ObjectRegistry } from '@happyvertical/smrt-core';
import { Person } from '@happyvertical/smrt-profiles';

describe('Person Model', () => {
  const testDbUrl = \`file::memory:test-person-\${Date.now()}-\${Math.random()}\`;
  let persons: any;

  beforeAll(async () => {
    persons = await ObjectRegistry.getCollection<typeof Person.prototype>('Person', {
      db: { url: testDbUrl, type: 'sqlite' },
    });
  });

  it('should create a person', async () => {
    const person = await persons.create({
      name: 'John Smith',
    });

    expect(person.id).toBeDefined(); // ❌ FAILS - person.id is null
    expect(person.name).toBe('John Smith');
  });

  it('should persist person to database and retrieve it', async () => {
    const person = await persons.create({
      name: 'Test Persistence',
    });

    const all = await persons.list({});
    const found = all.find((p: any) => p.id === person.id);
    expect(found).toBeDefined(); // ❌ FAILS - can't find the record
    expect(found.name).toBe('Test Persistence');
  });
});
```

### Test Output

```
Created person: John Smith (null)  ← ID is null!
📦 Loaded manifest for external package class: Person (0 fields, 0 methods)  ← Empty manifest!
[TEST] Found 0 Person profiles in database  ← Records not persisted
AssertionError: expected null to be defined
```

## Comparison: Working vs Broken

### ✅ Working: Local STI Class (Council)

```typescript
// In praeco/src/models/Council.ts
@smrt()
export class Council extends Organization {
  meetingsUrl?: Meta<string>;
  timezone?: Meta<string>;
}

// Tests pass:
const council = await councils.create({ name: 'Test Council' });
console.log(council.id); // ✅ "clz1234..."
```

**Manifest loads correctly**:
```
📦 Loaded manifest for class: Council (9 fields, 2 methods)
```

### ❌ Broken: External Package STI Class (Person)

```typescript
// In smrt-profiles/src/Person.ts
@smrt()
export class Person extends Profile {
  // ... fields
}

// Imported in praeco:
import { Person } from '@happyvertical/smrt-profiles';

// Tests fail:
const person = await persons.create({ name: 'John Smith' });
console.log(person.id); // ❌ null
```

**Manifest doesn't load**:
```
📦 Loaded manifest for external package class: Person (0 fields, 0 methods)
```

## Impact

This affects **all STI classes from smrt-profiles** used in consuming packages:
- `Person`
- `Organization` 
- `Profile`
- `Bot`

Any praeco features that create Person profiles, CouncilMembers, or other profile-based models will fail.

## Root Cause Analysis

The manifest loader appears to handle local classes differently from external package classes:

1. **Local classes** (defined in consuming package):
   - Manifest includes field definitions from the class and all parent classes
   - IDs auto-generate correctly
   - Database operations work

2. **External classes** (from `@happyvertical/smrt-*` packages):
   - Manifest shows "0 fields, 0 methods"
   - No field definitions loaded
   - ID generation fails (returns null)
   - Database operations fail

**Hypothesis**: The manifest loader may not be following imports correctly to load manifests from external packages, or external package manifests aren't being generated/included in the published packages.

## Expected Behavior

External package STI classes should work identically to local STI classes:

1. Manifest should load with all fields from class + parent classes
2. `create()` should auto-generate IDs
3. Database persistence should work correctly
4. `findById()` and `list()` should retrieve records

## Environment

- **SMRT Version**: Latest main (as of 2025-11-17)
- **Package**: `praeco` consuming `@happyvertical/smrt-profiles`
- **Database**: SQLite (`:memory:` databases)
- **Test Framework**: vitest with `smrt test`

--- a/packages/core/src/registry.ts
+++ b/packages/core/src/registry.ts
@@ -1488,6 +1488,11 @@ export class ObjectRegistry {
         registered.packageName = manifestEntry.packageName;
       }
 
+      // Store extends info for getAllFields() to use (if present)
+      if (manifestEntry.extends) {
+        registered.extends = manifestEntry.extends;
+      }
+
       console.log(
         `📦 Loaded manifest for external package class: ${className} (${registered.fields.size} fields, ${registered.methods.size} methods)`,
       );

